stock_storage_type/models/stock_location.py

```diff
@@ -141,21 +141,48 @@ def _compute_leaf_location_ids(self):
             leaves = self.search([("id", "in", leave_ids)])
             loc.leaf_location_ids = leaves
 
-    @api.depends("quant_ids", "in_move_ids", "in_move_line_ids")
+    def _should_compute_will_contain_product_ids(self):
+        return any(
+            location.do_not_mix_products
```

Comment:
No need to compute if "do not mix lot" is used as the product cache table won't be used but the lot cache table
```suggestion
            location.do_not_mix_products and not location.do_not_mix_lots

```

Reply:
I'll change the domain (https://github.com/OCA/wms/blob/5a44b7c04fcbe283ae0b36dbc3653769f127fb6b/stock_storage_type/models/stock_location_storage_type.py#L138-L148) the same way otherwise it is not obvious to understand why the field is empty but still in the domain.

Reply:
@jbaudoux after the change, tests do not pass, and I think it's normal: even if a location do not mix lots, some products may not be using lots, so we still need to cache the products and check for both.

Reply:
Then we could store only products tracked by lot in one case and those not in the other... but I don't think it's worth it re. added complexity vs the gain.

Reply:
@guewen Good point